### DISCUSSION TITLE
Child spans for "fire and forget" remote work?

### DISCUSSION THREAD

**Discussion by ProductiveRage** [jaegertracing/jaeger#3413] [Q&A] [answered]
I'm trying to confirm whether what I want to do is possible with Jaeger or not.. I want to have a trace that describes a process that calls several web endpoints in a "fire and forget" manner, meaning that it doesn't wait for their responses - but I'd still like for the work in those endpoints to be recorded as child spans of the trace. This would mean that the total time for the parent process' span would end before the child spans of the web endpoints finished and I've never seen any examples of Jaeger traces where that is the case.

I know that this can work if I _do_ wait for the web endpoints to respond because that will keep the parent span "open" but that's not the model that I'd like to use.

In case it makes a difference, I'm writing my code in C# and using the [OpenTelemetry-dotnet](https://github.com/open-telemetry/opentelemetry-dotnet) library.

Any pointers would be much appreciated - even if it is only definitive information that this sort of model is not a good fit for Jaeger tracing!

**Comment by yurishkuro** [comment, marked as answer]
It's perfectly fine for parent span to finish before children. This is how std OpemTracing instrumentation for Kafka works. I have a chapter in my book on that, with code example on GitHub.

**Reply by ProductiveRage** [reply]
That's great to hear! Can you point me to one of the example in your GitHub code, please? I had a quick look at the list of repositories for your account but I wasn't sure where to start.

**Reply by yurishkuro** [reply]
https://github.com/PacktPublishing/Mastering-Distributed-Tracing/tree/master/Chapter05

**Reply by ProductiveRage** [reply]
Thanks for that, I'll try to see if I can find what I need in there!

(I tried to do what as I described with the C# client but couldn't make it work, which made me fear that I was going against the grain of how Jaeger is intended to work - maybe I was doing it wrong or maybe there is an issue in the C# library but just knowing that it _should_ be possible is very helpful)

**Reply by yurishkuro** [reply]
Here's a trace discussed in the book that shows the interactions where children start after parent ends.

![image](https://user-images.githubusercontent.com/3523016/143780155-c2150636-0d72-4277-aef4-80bb9fefa5e0.png)
